app/services/action_validator.rb

```diff
@@ -0,0 +1,39 @@
+class ActionValidator
+
+  def initialize(params)
+    @params = params
+  end
+
+  def form
+    @form ||= Form.includes(:form_elements).find(@params[:form_id])
+  end
+
+  def valid?
+    errors.size == 0
+  end
+
+  def errors
+    errors = []
+    form.form_elements.each do |element|
+      error = validate_field(element)
+      errors.push(error) unless error.nil?
+    end
+    errors
+  end
+
+  def validate_field(form_element)
+    key = form_element.name.to_sym
+    if form_element.required == true
```

Comment:
Being a predicate, `if form_element.required` is sufficient. For maximum readability, you could use `.required?`, which AR defines for you as standard.